app/src/main/java/com/github/geohunt/app/model/database/firebase/FirebaseClaim.kt

```diff
@@ -0,0 +1,59 @@
+package com.github.geohunt.app.model.database.firebase
+
+import android.graphics.Bitmap
+import com.github.geohunt.app.model.BaseLazyRef
+import com.github.geohunt.app.model.LazyRef
+import com.github.geohunt.app.model.database.api.Challenge
+import com.github.geohunt.app.model.database.api.Claim
+import com.github.geohunt.app.model.database.api.Location
+import com.github.geohunt.app.model.database.api.User
+import com.github.geohunt.app.utility.*
+import com.google.android.gms.tasks.Task
+import java.time.LocalDateTime
+
+data class FirebaseClaim(
+    override val id: String,
+    override val challenge: LazyRef<Challenge>,
+    override val user: LazyRef<User>,
+    override val time: LocalDateTime,
+    override val location: Location,
+) : Claim
+
+class FirebaseClaimRef(
+    override val id: String,
+    private val database: FirebaseDatabase
+) : BaseLazyRef<Claim>() {
+
+    override fun fetchValue(): Task<Claim> {
+        val coarseHash = id.substring(0, Location.COARSE_HASH_SIZE)
+        val elementId = id.substring(Location.COARSE_HASH_SIZE)
```

Comment:
I do not understand why these two IDs are not separated, both in the code and in the database.

Reply:
The IDs are separated because the actual claim ID (which is the argument of `FirebaseClaimRef` ) is the combination of both coarseLocation and a unique ID. This is to make it consistent with the Challenge implementation where we decided that we are going to group the challenges with their coarse location and have a unique ID+coarseLocationID as the challengeID. So essentially it makes the grouping similar to the challenge table in our database.

Reply:
Okay, makes sense now, thank you !